tests/publiccloud/run_ltp.pm

```diff
@@ -145,7 +145,11 @@ sub run {
     assert_script_run("git clone -q --single-branch -b $kirk_branch --depth 1 $kirk_repo");
     $instance->run_ssh_command(cmd => 'sudo CREATE_ENTRIES=1 ' . get_ltproot() . '/IDcheck.sh', timeout => 300);
     record_info('Kernel info', $instance->run_ssh_command(cmd => q(rpm -qa 'kernel*' --qf '%{NAME}\n' | sort | uniq | xargs rpm -qi)));
-    record_info('VM Detect', $instance->run_ssh_command(cmd => 'systemd-detect-virt'));
+    if (get_var('PUBLIC_CLOUD_INSTANCE_TYPE') =~ /-metal$/) {
+        record_info('VM type', $instance->run_ssh_command(cmd => '! systemd-detect-virt');
+    } else {
+        record_info('VM type', $instance->run_ssh_command(cmd => 'systemd-detect-virt'));
+    }
```

Comment:
simpler like:
```suggestion
    my $cmd;
    $cmd = "! " if (get_var('PUBLIC_CLOUD_INSTANCE_TYPE') =~ /-metal$/);
    record_info('VM type', $instance->run_ssh_command(cmd => $cmd . 'systemd-detect-virt'));
```

Reply:
Not sure if here escape needed, like "\! " 🙂

Reply:
> I'd suggest simpler, like:

I find my version more aesthetically pleasing as code can be read both horizontally and vertically. :)

And we define `$cmd` variable later in this module.

Reply:
it is too much redundant, repeating a big amout of code for a simple "!".

Reply:
A kind post-comment: being info-reading, we could have more simply proceeded on failure, like: `run_ssh_command(cmd => $cmd, proceed_on_failure => 1)`

Reply:
> A kind post-comment: being info-reading, we could have more simply proceeded on failure, like: `run_ssh_command(cmd => $cmd, proceed_on_failure => 1)`

We want `systemd-detect-virt` to fail only on bare-metal.  We are both asserting and showing information here.

Reply:
ok